modules/services/window-managers/bspwm/options.nix

```diff
@@ -0,0 +1,220 @@
+        example = "south";
+      };
+
+      splitRatio = mkOption {
+        type = types.nullOr types.float;
+        default = null;
+        description = "The ratio between the new window and the previous existing window in the desktop";
+        example = 0.65;
+      };
+
+      hidden = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the node isn't going to occupy any space";
+        example = true;
+      };
+
+      sticky = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the node is going to stay in the focused desktop of its monitor";
+        example = true;
+      };
+
+      private = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the node is going to try to stay in the same tiling position and size";
+        example = true;
+      };
+
+      locked = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the node is going to ignore the 'node --close' messae";
+        example = true;
+      };
+
+      marked = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the node is going to be marked for deferred actions";
+        example = true;
+      };
+
+      center = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the node will be put in the center of the screen in floating mode.";
+        example = true;
+      };
+
+      follow = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the previous focused node is going to stay focused";
+        example = true;
+      };
+
+      manage = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If set to false, the window will not be managed by bspwm at all";
+        example = true;
+      };
+
+      focus = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the new node is going to gain the focus";
+        example = true;
+      };
+
+      border = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "If it's set to true, the new node is going to have border";
+        example = true;
+      };
+    };
+  };
+
+in
+
+{
+  xsession.windowManager.bspwm = {
+    enable = mkEnableOption "bspwm window manager.";
+
+    package = mkOption {
+        type = types.package;
+        default = pkgs.bspwm;
+        defaultText = "pkgs.bspwm";
+        description = "bspwm package to use.";
+        example = "pkgs.bspwm-unstable";
+    };
+
+    config = mkOption {
```

Comment:
Instead of `config`, this should be called `settings` for consistency and avoiding the heavily overloaded `config` variable name.